refactor(linter/plugins): simplify `comments` getter

overlookmotel opened the pull request:
Follow-on after #14727.

Since we now have to call into deserializer module to update `astId` at end of linting a file, we may as well reset the buffers at that point too. This allows simplifying the `comments` getter.

overlookmotel commented on crates/oxc_ast/src/serialize/mod.rs:
```diff
@@ -121,14 +121,7 @@ impl Program<'_> {
-    let refUint32 = uint32;
-    let refUint8 = uint8;
-    let refSourceText = sourceText;
     const localAstId = astId;
```
No, `astId` is always present in the source. It's the minifier that removes it if it's not used. So it's fine.

lilnasy reviewed (commented):
I haven't looked at other deserializers - I didn't get into them too deep - but this is nice. I like this a lot better.

overlookmotel committed:
refactor(linter/plugins): simplify `comments` getter (#14728)

Follow-on after #14727.

Since we now have to call into deserializer module to update `astId` at end of linting a file, we may as well reset the buffers at that point too. This allows simplifying the `comments` getter.
